Exercise_2.java

```diff
@@ -8,11 +8,29 @@
        of pivot */
     void swap(int arr[],int i,int j){
         //Your code here   
+        if(i<arr.length && j<arr.length) { // Checking the i and j around the max value that can fit in arr.

```

Comment:
Check for this condition. Would it ever be False?

Reply:
this will be always true , because we are swapping within in the array itself. i just wrote this because of my daily job , here we will get the lot of null pointer, we will always check for them like != null and aware of length condition to don't get the array index out of bound exceptions.

Reply:
Yup, here this check is redundant though. I understand it is there because of coding habit